cli/flox/src/commands/build.rs

```diff
@@ -23,15 +26,73 @@ pub struct Build {
     /// the 'build' table in the environment's manifest.toml.
     /// If not specified, all packages are built.
     #[bpaf(positional("build"))]
-    package: Vec<String>,
+    packages: Vec<String>,
 }
 
 impl Build {
     #[instrument(name = "build", skip_all)]
-    pub async fn handle(self, _config: Config, _flox: Flox) -> Result<()> {
+    pub async fn handle(self, config: Config, flox: Flox) -> Result<()> {
         subcommand_metric!("build");
 
-        message::plain("🚧 👷 heja, a new command is in construction here, stay tuned!");
+        if !config.features.unwrap_or_default().build {
+            message::plain("🚧 👷 heja, a new command is in construction here, stay tuned!");
+            bail!("'build' feature is not enabled.");
+        }
+
+        let env = self
+            .environment
+            .detect_concrete_environment(&flox, "Build")?;
+
+        if let ConcreteEnvironment::Remote(_) = &env {
+            bail!("Cannot build from a remote environment");
+        };
+
+        let mut env = env.into_dyn_environment();
+
+        let base_dir = env.parent_path()?;
+        let flox_env = env.activation_path(&flox)?;
+
+        let packages_to_build = make_packages_to_build(&env.lockfile(&flox)?, self.packages)?;
+
+        let builder = FloxBuildMk;
+        let output = builder.build(&base_dir, &flox_env, &packages_to_build)?;
+
+        for message in output {
+            match message {
+                Output::Stdout(line) => println!("{line}"),
+                Output::Stderr(line) => eprintln!("{line}"),
+                Output::Exit(status) if status.success() => {
+                    message::created("Build completed successfully");
+                    break;
+                },
+                Output::Exit(status) => {
+                    bail!("Build failed with status: {status}");
```

Comment:
suggestion: only bail if this is non-zero

Reply:
If the status signals success, we just break in the branch above